packages/clerk-js/src/ui/components/UserProfile/EnterpriseAccountsSection.tsx

```diff
@@ -15,58 +42,83 @@ export const EnterpriseAccountsSection = () => {
       centered={false}
     >
       <ProfileSection.ItemList id='enterpriseAccounts'>
-        {user?.samlAccounts.map(account => {
-          const label = account.emailAddress;
-          const providerName = getSamlProviderName(account.provider);
-          const providerLogoUrl = getSamlProviderLogoUrl(account.provider);
-          const error = account.verification?.error?.longMessage;
-
-          return (
-            <ProfileSection.Item
-              id='enterpriseAccounts'
-              sx={t => ({
-                gap: t.space.$2,
-                justifyContent: 'start',
-              })}
-              key={account.id}
-            >
-              <Image
-                elementDescriptor={[descriptors.providerIcon]}
-                elementId={descriptors.enterpriseButtonsProviderIcon.setId(account.provider)}
-                alt={providerName}
-                src={providerLogoUrl}
-                sx={theme => ({ width: theme.sizes.$4 })}
-              />
-              <Box sx={{ whiteSpace: 'nowrap', overflow: 'hidden' }}>
-                <Flex
-                  gap={2}
-                  center
-                >
-                  <Text
-                    truncate
-                    colorScheme='body'
-                  >
-                    {providerName}
-                  </Text>
-                  <Text
-                    truncate
-                    as='span'
-                    colorScheme='secondary'
-                  >
-                    {label ? `• ${label}` : ''}
-                  </Text>
-                  {error && (
-                    <Badge
-                      colorScheme='danger'
-                      localizationKey={localizationKeys('badge__requiresAction')}
-                    />
-                  )}
-                </Flex>
-              </Box>
-            </ProfileSection.Item>
-          );
-        })}
+        {user?.enterpriseAccounts.map(account => (
+          <EnterpriseAccount
+            key={account.id}
+            account={account}
+          />
+        ))}
       </ProfileSection.ItemList>
     </ProfileSection.Root>
   );
 };
+
+const EnterpriseAccountProviderIcon = ({ account }: { account: EnterpriseAccountResource }) => {
+  const { provider } = account;
+  const providerName = getEnterpriseAccountProviderName(account);
+
+  if (isOAuthBuiltInProvider(provider) || isSamlProvider(provider)) {
```

Comment:
We could simplify this by looking at `account.enterpriseConnection.logoPublicUrl`, but I'll wait until FAPI changes get deployed in order to validate this.